server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -2522,6 +2584,9 @@ public static IndexMetadata fromXContent(XContentParser parser, Map<String, Mapp
                         case KEY_TIMESTAMP_RANGE:
                             builder.timestampRange(IndexLongFieldRange.fromXContent(parser));
                             break;
+                        case KEY_EVENT_INGESTED_RANGE:
```

Comment:
Can we add a test for from xcontent that verifies that we get the right event ingested field and also that the bwc works when it is missing?

Reply:
Thanks.

I've added a simple test for event.ingested in `fromXContent` here: https://github.com/elastic/elasticsearch/pull/106252/commits/5b1e47bb41cba19b0632acd9ac7596511074457a

There are dozens of tests that rely on this behavior now (if you comment out this code lots of tests, including IT tests fail).

And for bwc, like my previous comment, I had to do various commits to get existing bwc tests passing with earlier versions of this PR. How would you recommend structuring a bwc test for this code? With XContent (vs. Writeables) I'm even less clear how to structure a bwc test here, so I welcome input on how to set that up (for instance, is there a known good example of doing this you can point at)?

Reply:
I would write the xcontent (like what is done in `IndexMetadataTests.testIndexMetadataSerialization`), strip the field out and then check that `fromXContent` can parse it. I prefer to have an explicit test for things like this, since it is very serious if this should not work (and the rolling upgrade tests etc that caught the issue so far may change in the future).

Reply:
Got it. Thanks. I have made an attempt at that in the [latest commit](https://github.com/elastic/elasticsearch/pull/106252/commits/56e4d1a384735c9017f8ae0daaa51baea4710c2f).

Reply:
:+1: looks good